### PR TITLE
Drop support for PyTorch 1.12

### PR DESCRIPTION
**Pull request by awaelchli**
## What does this PR do?

With support for the new PyTorch 2.2 version (#19289), we're removing PyTorch 1.12 support too keep our 4-version support window (1.13, 2.0, 2.1, 2.2).

Previous PR: #18691

----
📚 Documentation preview 📚: https://pytorch-lightning--19300.org.readthedocs.build/en/19300/

cc @borda @carmocca @justusschock @awaelchli

### CODE REVIEW

File: .github/checkgroup.yml [hunk 1]
```diff
@@ -19,18 +19,18 @@ subprojects:
       - "!*.md"
       - "!**/*.md"
     checks:
-      - "pl-cpu (macOS-11, lightning, 3.8, 1.12, oldest)"
-      - "pl-cpu (macOS-11, lightning, 3.9, 1.12)"
+      - "pl-cpu (macOS-11, lightning, 3.8, 1.13, oldest)"
+      - "pl-cpu (macOS-11, lightning, 3.9, 1.13)"
```
Comment (carmocca) [inline .github/checkgroup.yml:23]: I dont see these existing anymore after your CI updates

```suggestion
```

File: .github/checkgroup.yml [hunk 2]
```diff
@@ -19,18 +19,18 @@ subprojects:
...
       - "pl-cpu (macOS-11, lightning, 3.10, 1.13)"
       - "pl-cpu (macOS-11, lightning, 3.10, 2.0)"
       - "pl-cpu (macOS-11, lightning, 3.10, 2.1)"
-      - "pl-cpu (ubuntu-20.04, lightning, 3.8, 1.12, oldest)"
-      - "pl-cpu (ubuntu-20.04, lightning, 3.9, 1.12)"
+      - "pl-cpu (ubuntu-20.04, lightning, 3.8, 1.13, oldest)"
+      - "pl-cpu (ubuntu-20.04, lightning, 3.9, 1.13)"
```
Comment (carmocca) [inline .github/checkgroup.yml:28]:
```suggestion
```

File: .github/checkgroup.yml [hunk 3]
```diff
@@ -19,18 +19,18 @@ subprojects:
...
       - "pl-cpu (ubuntu-20.04, lightning, 3.10, 1.13)"
       - "pl-cpu (ubuntu-20.04, lightning, 3.10, 2.0)"
       - "pl-cpu (ubuntu-20.04, lightning, 3.10, 2.1)"
-      - "pl-cpu (windows-2022, lightning, 3.8, 1.12, oldest)"
-      - "pl-cpu (windows-2022, lightning, 3.9, 1.12)"
+      - "pl-cpu (windows-2022, lightning, 3.8, 1.13, oldest)"
+      - "pl-cpu (windows-2022, lightning, 3.9, 1.13)"
```
Comment (carmocca) [inline .github/checkgroup.yml:33]:
```suggestion
```

File: .github/checkgroup.yml [hunk 4]
```diff
@@ -193,18 +191,18 @@ subprojects:
       - "!*.md"
       - "!**/*.md"
     checks:
-      - "fabric-cpu (macOS-11, lightning, 3.8, 1.12, oldest)"
-      - "fabric-cpu (macOS-11, lightning, 3.9, 1.12)"
+      - "fabric-cpu (macOS-11, lightning, 3.8, 1.13, oldest)"
+      - "fabric-cpu (macOS-11, lightning, 3.9, 1.13)"
```
Comment (carmocca) [inline .github/checkgroup.yml:195]:
```suggestion
```

File: .github/checkgroup.yml [hunk 5]
```diff
@@ -193,18 +191,18 @@ subprojects:
...
       - "fabric-cpu (macOS-11, lightning, 3.10, 1.13)"
       - "fabric-cpu (macOS-11, lightning, 3.10, 2.0)"
       - "fabric-cpu (macOS-11, lightning, 3.11, 2.1)"
-      - "fabric-cpu (ubuntu-20.04, lightning, 3.8, 1.12, oldest)"
-      - "fabric-cpu (ubuntu-20.04, lightning, 3.9, 1.12)"
+      - "fabric-cpu (ubuntu-20.04, lightning, 3.8, 1.13, oldest)"
+      - "fabric-cpu (ubuntu-20.04, lightning, 3.9, 1.13)"
```
Comment (carmocca) [inline .github/checkgroup.yml:200]:
```suggestion
```

File: .github/checkgroup.yml [hunk 6]
```diff
@@ -193,18 +191,18 @@ subprojects:
...
       - "fabric-cpu (ubuntu-20.04, lightning, 3.10, 1.13)"
       - "fabric-cpu (ubuntu-20.04, lightning, 3.10, 2.0)"
       - "fabric-cpu (ubuntu-20.04, lightning, 3.11, 2.1)"
-      - "fabric-cpu (windows-2022, lightning, 3.8, 1.12, oldest)"
-      - "fabric-cpu (windows-2022, lightning, 3.9, 1.12)"
+      - "fabric-cpu (windows-2022, lightning, 3.8, 1.13, oldest)"
+      - "fabric-cpu (windows-2022, lightning, 3.9, 1.13)"
```
Comment (carmocca) [inline .github/checkgroup.yml:205]:
```suggestion
```

File: .github/workflows/docker-build.yml [hunk 7]
```diff
@@ -104,7 +103,6 @@ jobs:
         include:
           # These are the base images for PL release docker images,
           # so include at least all of the combinations in release-dockers.yml.
```
Comment (carmocca) [inline .github/workflows/docker-build.yml:105]: Did you update rlease-dockers.yml as this comment suggests?

File: tests/tests_pytorch/strategies/test_common.py [hunk 8]
```diff
@@ -54,7 +54,6 @@ def test_evaluate(tmpdir, trainer_kwargs):
     torch.testing.assert_close(old_weights, new_weights)
 
 
-@RunIf(min_torch="1.13")
```
Comment (carmocca) [inline tests/tests_pytorch/strategies/test_common.py:57]: Did you search for `max_torch="1.12"` too?

**Review by carmocca — APPROVED** [review]

File: tests/tests_pytorch/strategies/test_common.py (hunk 8 above)
Comment (awaelchli) [reply tests/tests_pytorch/strategies/test_common.py:57]: Yes there don't appear to be any

File: .github/workflows/docker-build.yml (hunk 7 above)
Comment (carmocca) [reply .github/workflows/docker-build.yml:105]: @Borda must have renamed it at some point. Any suggestions Jirka?

File: tests/tests_pytorch/strategies/test_common.py (hunk 8 above)
Comment (Borda) [reply tests/tests_pytorch/strategies/test_common.py:57]: docker workflows were merged into one: https://github.com/Lightning-AI/pytorch-lightning/blob/master/.github/workflows/docker-build.yml

File: .github/workflows/docker-build.yml (hunk 7 above)
Comment (Borda) [reply .github/workflows/docker-build.yml:105]: docker workflows were merged into one: https://github.com/Lightning-AI/pytorch-lightning/blob/master/.github/workflows/docker-build.yml

File: docs/source-pytorch/versioning.rst [hunk 9]
```diff
@@ -81,6 +81,12 @@ The table below indicates the coverage of tested versions in our CI. Versions ou
      - ``torch``
      - ``torchmetrics``
      - Python
+   * - 2.2
+     - 2.2
+     - 2.2
+     - ≥1.13, ≤2.2
+     - ≥0.7.0
+     - ≥3.8, ≤3.11
```
Comment (Borda) [inline docs/source-pytorch/versioning.rst:89]:
```suggestion
```
This is not part of this PR, right?

**Review by Borda — APPROVED** [review]

File: .github/workflows/docker-build.yml (hunk 7 above)
Comment (awaelchli) [reply .github/workflows/docker-build.yml]: Ok, since they were combined into the same image, I updated the comment. 

While we are here, I think L77 should also be changed to tag the latest correctly?

File: docs/source-pytorch/versioning.rst (hunk 9 above)
Comment (awaelchli) [reply docs/source-pytorch/versioning.rst:89]: Why not? We want to document with which versions of PyTorch Lightning 2.2 is compatible with.

Comment (Borda) [reply docs/source-pytorch/versioning.rst:89]: Yes, I am just saying it shall be in the other PR =)

**Review by lantiga — APPROVED** [review]
